enhancements/etcd/etcd-size-tuning.md

```diff
@@ -0,0 +1,236 @@
+* Currently, etcd can not detect whether its backend quota configuration value could lead to a database that's larger than the available storage.
+  * We could potentially add a prometheus alert to fire when the etcd node is running out of disk space.
+  * We could potentially add some validation to the etcd server to check the quota configuration value against the space available and error if too small.
+    * This enhancement would need to take into consideration multiple instances sharing a disk to reduce the likelihood of each individual instance thinking it has enough space, but not realizing the other instances are also vying for that space.
+* Etcd puts out a warning if the backend quota is configured to be larger than 8GiB and the documentation calls out that 8GiB is the suggested maximum for normal clusters.
+  * We've chosen to allow values higher than this soft-limit as the customers/clusters asking for this feature are larger than would be considered a "normal" cluster.
+  * [The Perf-Scale team has tested values](https://issues.redhat.com/browse/PERFSCALE-2881) up to 40GiB and found that instability was introduced around 34GiB which influenced the recommended maximum: 32GiB.
+
+### Drawbacks
+
+* There will be a required etcd rollout when changing values, this happens automatically by the Cluster Etcd Operator to re-render the etcd podspec with the new backend quota value.
+
+## Open Questions [optional]
+
+For each of these, we would need to compare against the current default of 8GiB. Increasing the quota does not necessarily increase the following, but at the extreme case (e.g. increase to 16GiB and using nearly that much) we expect the following to perform differently.
+* How long does it take to sync a snapshot of a given size to another peer; for reboot/recovery?
+* How long does it take to compact & defrag a database of a given size?
+  * Defragmentation will pause writes to etcd, reads from etcd will be unaffected.
+
+## Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+We will need to test a cluster with a larger quota to ensure everything works as expected.
+We should also load the database with a larger quota and make note of the memory requirements of a defragmentation.
+  - We should do this with some values we could assume a customer would use; 16GiB, 32GiB, 100GiB (in the extreme).
+We need to test that it is not possible to shrink the quota once it has be set to a larger value.
+
+## Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+None
+
+### Dev Preview -> Tech Preview
+
+N/A
+
+### Tech Preview -> GA
+
+- Known issue in API Server
+  - When the API server restarts, it's currently possible for all watchers to be reset at the same time and request a large amount of data from etcd, leading to a crash-loop in the API server.
+  - [https://github.com/kubernetes/enhancements/pull/4453](API Streaming)
```

Comment:
I'd suggest pointing to the enhancement tracking issue which is https://github.com/kubernetes/enhancements/issues/3157